astropy/coordinates/sky_coordinate.py

```diff
@@ -72,6 +72,9 @@ def _represent_as_dict(self):
         if 'distance' in attrs and np.all(obj.distance == 1.0):
             attrs.remove('distance')
 
+        if 's' in obj.frame.data.differentials:
```

Comment:
This looks just right. It might not be bad to future-proof a little by warning is there is anything else in `differentials`.

Reply:
I'm glad it looks right, but it doesn't work! 😛 

But, as you said above, maybe `skycoord_equal` is not the best test here? Maybe `SkyCoord` should implement its own `__eq__`? That sounds like a pain, though.

Reply:
We've had long discussions about `SkyCoord.__eq__` already - let's just have a working function for testing purposes for now...

Reply:
:+1: on a *working* function.  And while you're at it why not move that function to a rational place like into the coordinates subpackage (`test_sky_coord.py`?).